### PR TITLE
Issue 18530

### PR DESCRIPTION
**Pull request by erickgonzalez**
New endpoint `/api/v2/languages/{language}/i18n` to get all the messages, for the path param you can pass the languageId or the languageCode + CountryCode. 

If you want to get the messages for the default lang, just pass `default`

LanguageUtilTest was not added to the MainSuite when created, now that is added found that this test was failing `getLanguageId_existing_base_expected_non_negative_lang_Test` because another test creates the same languages. Fixing it by checking if the lang already exists before creating.

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguagesResource.java [hunk 1]
```diff
@@ -184,6 +188,38 @@ public Response getMessages(@Context HttpServletRequest request,
...
+     */
+    @GET
+    @Path("/i18n/all")
+    @JSONP
+    @NoCache
+    @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
+    public Response getAllMessages (
+            @Context final HttpServletRequest request,
+            @Context final HttpServletResponse response,
+            @DefaultValue("-1") @QueryParam("language") final String language){
```
Comment (dsilvam) [inline dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguagesResource.java:208]: "language" shouldn't be a query parameter. 
Pathing should be
`v2/languages/{language}/i18n` to get all the messages for a language

Comment (dsilvam) [reply dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguagesResource.java:208]: Requesting changes above

**Review by dsilvam — CHANGES_REQUESTED** [review]

**Review by freddyucv — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguagesResource.java (hunk 1 above)
Comment (fmontes) [reply dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguagesResource.java:208]: The lang is optional, that's why is a queryparam.

Comment (erickgonzalez) [reply dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguagesResource.java]: done

File: dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguagesResource.java [hunk 2]
```diff
@@ -23,12 +23,16 @@
...
+import com.liferay.portal.language.LanguageUtil;
 import com.liferay.portal.model.User;
 import com.rainerhahnekamp.sneakythrow.Sneaky;
 import java.util.List;
+import java.util.Locale;
+import java.util.Map;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 import javax.ws.rs.DELETE;
+import javax.ws.rs.DefaultValue;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguagesResource.java:35]: Codacy Issue found: [Avoid unused imports such as 'javax.ws.rs.DefaultValue'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5588495)

Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguagesResource.java:35]: Codacy Issue found: [Avoid unused imports such as 'javax.ws.rs.DefaultValue'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5588495)

Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguagesResource.java:35]: Codacy Issue found: [Avoid unused imports such as 'javax.ws.rs.DefaultValue'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5588495)

File: dotCMS/src/integration-test/java/com/liferay/portal/language/LanguageUtilTest.java [hunk 3]
```diff
@@ -70,12 +73,18 @@ public void getLanguageId_existing_expected_non_negative_lang_Test() throws DotS
     }
 
     @Test
-    public void getLanguageId_existing_base_expected_non_negative_lang_Test() throws DotSecurityException, DotDataException, SystemException {
-
-        final Language baseLanguage = new LanguageDataGen().languageCode("fr").countryCode(null).nextPersisted();
-        final Language frLanguage   = new LanguageDataGen().languageCode("fr").countryCode("FR").nextPersisted();
+    public void getLanguageId_existing_base_expected_non_negative_lang_Test() {
+
+        final Language baseLanguage = (UtilMethods.isSet(APILocator.getLanguageAPI().getLanguage("fr",null))) ?
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/liferay/portal/language/LanguageUtilTest.java:78]: Codacy Issue found: [Useless parentheses.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5588495)

File: dotCMS/src/integration-test/java/com/liferay/portal/language/LanguageUtilTest.java [hunk 4]
```diff
@@ -70,12 +73,18 @@ public void getLanguageId_existing_expected_non_negative_lang_Test() throws DotS
...
+                APILocator.getLanguageAPI().getLanguage("fr",null) :
+                new LanguageDataGen().languageCode("fr").countryCode(null).nextPersisted();
+        final Language frLanguage   = (UtilMethods.isSet(APILocator.getLanguageAPI().getLanguage("fr","fr"))) ?
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/liferay/portal/language/LanguageUtilTest.java:81]: Codacy Issue found: [Useless parentheses.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5588495)

Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/liferay/portal/language/LanguageUtilTest.java:81]: Codacy Issue found: [Useless parentheses.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5588495)

File: dotCMS/src/integration-test/java/com/liferay/portal/language/LanguageUtilTest.java (hunk 3 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/liferay/portal/language/LanguageUtilTest.java:78]: Codacy Issue found: [Useless parentheses.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5588495)

File: dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguagesResource.java (hunk 2 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguagesResource.java:35]: Codacy Issue found: [Avoid unused imports such as 'javax.ws.rs.DefaultValue'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5588495)

Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguagesResource.java:35]: Codacy Issue found: [Avoid unused imports such as 'javax.ws.rs.DefaultValue'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5588495)

File: dotCMS/src/integration-test/java/com/liferay/portal/language/LanguageUtilTest.java (hunk 3 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/liferay/portal/language/LanguageUtilTest.java:78]: Codacy Issue found: [Useless parentheses.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5588495)

File: dotCMS/src/integration-test/java/com/liferay/portal/language/LanguageUtilTest.java (hunk 4 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/liferay/portal/language/LanguageUtilTest.java:81]: Codacy Issue found: [Useless parentheses.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5588495)

**Review by nollymar — APPROVED** [review]

File: dotCMS/src/integration-test/java/com/liferay/portal/language/LanguageUtilTest.java [hunk 5]
```diff
@@ -70,12 +73,18 @@ public void getLanguageId_existing_expected_non_negative_lang_Test() throws DotS
     }
 
     @Test
-    public void getLanguageId_existing_base_expected_non_negative_lang_Test() throws DotSecurityException, DotDataException, SystemException {
-
-        final Language baseLanguage = new LanguageDataGen().languageCode("fr").countryCode(null).nextPersisted();
-        final Language frLanguage   = new LanguageDataGen().languageCode("fr").countryCode("FR").nextPersisted();
+    public void getLanguageId_existing_base_expected_non_negative_lang_Test() {
```
Comment (dsilvam) [inline dotCMS/src/integration-test/java/com/liferay/portal/language/LanguageUtilTest.java:76]: missing `test` in method name

**Review by jdotcms — APPROVED** [review]

File: dotCMS/src/integration-test/java/com/liferay/portal/language/LanguageUtilTest.java (hunk 5 above)
Comment (erickgonzalez) [reply dotCMS/src/integration-test/java/com/liferay/portal/language/LanguageUtilTest.java]: done

**Review by dsilvam — APPROVED** [review]

File: dotCMS/src/integration-test/java/com/liferay/portal/language/LanguageUtilTest.java [hunk 6]
```diff
@@ -69,31 +77,61 @@ public void getLanguageId_existing_expected_non_negative_lang_Test() throws DotS
         APILocator.getLanguageAPI().deleteLanguage(language);
     }
 
+    private static class testCaseLanguageLocale{
```
Comment (dsilvam) [inline dotCMS/src/integration-test/java/com/liferay/portal/language/LanguageUtilTest.java:80]: start the class name with Uppercase

Comment (erickgonzalez) [reply dotCMS/src/integration-test/java/com/liferay/portal/language/LanguageUtilTest.java]: done

File: dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguagesResource.java (hunk 2 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguagesResource.java:35]: Codacy Issue found: [Avoid unused imports such as 'javax.ws.rs.DefaultValue'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5588495)

File: dotCMS/src/integration-test/java/com/liferay/portal/language/LanguageUtilTest.java [hunk 7]
```diff
@@ -69,31 +76,61 @@ public void getLanguageId_existing_expected_non_negative_lang_Test() throws DotS
...
+        }
+    }
+
+    @DataProvider
+    public static Object[] testCasesLanguageLocale() throws Exception {
+
+        // Setting web app environment
+        IntegrationTestInitService.getInstance().init();
+
+        final Language baseLanguage = (UtilMethods.isSet(APILocator.getLanguageAPI().getLanguage("fr",null))) ?
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/liferay/portal/language/LanguageUtilTest.java:95]: Codacy Issue found: [Useless parentheses.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5588495)

File: dotCMS/src/integration-test/java/com/liferay/portal/language/LanguageUtilTest.java [hunk 8]
```diff
@@ -69,31 +76,61 @@ public void getLanguageId_existing_expected_non_negative_lang_Test() throws DotS
...
+                APILocator.getLanguageAPI().getLanguage("fr",null) :
+                new LanguageDataGen().languageCode("fr").countryCode(null).nextPersisted();
+        final Language frLanguage   = (UtilMethods.isSet(APILocator.getLanguageAPI().getLanguage("fr","fr"))) ?
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/liferay/portal/language/LanguageUtilTest.java:98]: Codacy Issue found: [Useless parentheses.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5588495)

File: dotCMS/src/integration-test/java/com/liferay/portal/language/LanguageUtilTest.java [hunk 9]
```diff
@@ -69,31 +76,61 @@ public void getLanguageId_existing_expected_non_negative_lang_Test() throws DotS
...
+        String localeRequested;
+        long expectedLangId;
+
+        TestCaseLanguageLocale(final String localeRequested,final long expectedLangId){
+            this.localeRequested = localeRequested;
+            this.expectedLangId = expectedLangId;
+        }
+    }
+
+    @DataProvider
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/liferay/portal/language/LanguageUtilTest.java:89]: Codacy Issue found: [JUnit 4 tests that execute tests should use the @Test annotation, JUnit 5 tests should use @Test, @RepeatedTest, @TestFactory, @TestTemplate or @ParameterizedTest](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5588495)

**Pull request merged**